### PR TITLE
Build with TS 2.7

### LINKED ISSUE #42853: Adopt TS 2.7 for building VS Code
Tracks using VS Code 2.7 to build vscode.

May require a few fixes in built-in extensions that use `"strict": true` since TS added a few more checks here

### PR DESCRIPTION
**Pull request by mjbvz**
Fixes #42853

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -5636,9 +5636,9 @@ typescript-formatter@4.0.1:
     editorconfig "^0.13.2"
     glob-expand "^0.2.1"
 
-typescript@2.6.1:
-  version "2.6.1"
-  resolved "https://registry.yarnpkg.com/typescript/-/typescript-2.6.1.tgz#ef39cdea27abac0b500242d6726ab90e0c846631"
+typescript@2.7.1:
```
Comment (andy-ms) [inline yarn.lock]: Why was this changed but `typescript-formatter` wasn't?

Comment (andy-ms) [reply yarn.lock]: Running `yarn` on my checkout of this branch causes many `yarn.lock` changes including `typescript-formatter`.

Comment (mjbvz) [reply yarn.lock]: Bad rebase. The error is from TS 2.7.2 with ts-formatter 7.1.0 though

**Comment by mjbvz** [comment]
Updating yarn versions fixed the tslint error. Just a few formatting errors now. Thanks for the suggestion @andy-ms!
